### PR TITLE
fix(griffin) - add no-op groupby. fixed #464 

### LINKED ISSUE #464: support noop group by clause
This SQL errors out unnecessarily:

```sql
select x, sum(y) from tab group by x
```

`group by x` must be consumed and validated to allow friction free user experience

### PR DESCRIPTION
**Pull request by jaugsburger**

### CODE REVIEW

File: core/src/test/java/io/questdb/griffin/SqlParserTest.java [hunk 1]
```diff
@@ -4974,6 +4974,104 @@ public void testEmptyWhere() throws Exception {
...
+                        "    bid int,\n" +
+                        "    ask int\n" +
+                        ")  partition by NONE",
+                0,
+                "group by column does not match key column is select statement "
+        );
+    }
+
+    @Test
+    public void testNoopGroupByWith2Syms() throws Exception {
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/griffin/SqlParserTest.java:5062]: This must not work, but it does:
```
"select sym1, sym2, avg(bid) avgBid from x where sym1 in ('AA', 'BB' ) group by sym2"
```

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 2]
```diff
@@ -42,42 +42,172 @@
...
+                model.addGroupBy(n.token);
+
+                tok = optTok(lexer);
+            } while (tok != null && Chars.equals(tok, ','));
+        }
+
+        // expect [order by]
+
+        if (tok != null && isOrderKeyword(tok)) {
+            expectTok(lexer, "by");
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlParser.java:163]: Can we also  put "by" check as sql keyword? Across all clauses pls

File: core/src/test/java/io/questdb/griffin/SqlParserTest.java (hunk 1 above)
Comment (jaugsburger) [reply core/src/test/java/io/questdb/griffin/SqlParserTest.java:5077]: just added a test to assertFailure on that query and the test passes

File: core/src/test/java/io/questdb/griffin/SqlParserTest.java [hunk 3]
```diff
@@ -5044,6 +5044,21 @@ public void testNoopGroupByFailureWhenUsing2KeysInSelectStatementButOnlyOneInGro
         );
     }
 
+    @Test
+    public void testNoopGroupByFailureWhenUsing2KeysInSelectStatementButOnlyOneInGroupByV2() throws Exception {
+        assertFailure(
+                "select sym1, sym2, avg(bid) avgBid from x where sym1 in ('AA', 'BB' ) group by sym2",
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/griffin/SqlParserTest.java:5050]: may be tests pass all the time then? :)

i just modified this test:

```java
    @Test
    public void testNoopGroupByWith2Syms() throws Exception {
        assertQuery(
                "select-group-by sym1, sym2, avg(bid) avgBid from (select [sym1, sym2, bid] from x timestamp (ts) where sym1 in ('AA','BB'))",
                "select sym1, sym2, avg(bid) avgBid from x where sym1 in ('AA', 'BB' ) group by sym1",
                modelOf("x")
                        .col("sym1", ColumnType.SYMBOL)
                        .col("sym2", ColumnType.SYMBOL)
                        .col("bid", ColumnType.INT)
                        .col("ask", ColumnType.INT)
                        .timestamp("ts")
        );
    }

```

and it also passes. Why is that?

File: core/src/main/java/io/questdb/griffin/SqlParser.java (hunk 2 above)
Comment (jaugsburger) [reply core/src/main/java/io/questdb/griffin/SqlParser.java]: done

File: core/src/test/java/io/questdb/griffin/SqlParserTest.java (hunk 3 above)
Comment (jaugsburger) [reply core/src/test/java/io/questdb/griffin/SqlParserTest.java:5050]: ok, issue here is that SqlParserTest.assertQuery() does not call SqlCodeGenerator , which is where the exception is thrown. I'll fix the tests now

**Pull request closed without merging**